### PR TITLE
[WIP] Set nightly builds for complex mode

### PR DESCRIPTION
**Pull request by IgorBaratta**
Proposal:
 - nightly builds that run every day at 00:00 am UTC,  to test complex mode.
 - reuse code (steps) for the real and complex workflows.

Real mode (default) is triggered when pushing to any branch.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -148,4 +202,14 @@ workflows:
           filters:
             branches:
               only:
-                - master
+                - igor/nightly-build
```
Comment (blechta) [inline .circleci/config.yml]: Why is this changed?

File: .circleci/config.yml [hunk 2]
```diff
@@ -148,4 +202,14 @@ workflows:
...
+  nightly:
+    triggers:
+      - schedule:
+          cron: "0 0 * * *"
+          filters:
+            branches:
+              only:
+                - igor/nightly-build
```
Comment (blechta) [inline .circleci/config.yml]: Should be changed to `master` when merged? Or could we run all branches?

File: .circleci/config.yml [hunk 3]
```diff
@@ -148,4 +202,14 @@ workflows:
           filters:
             branches:
               only:
-                - master
+                - igor/nightly-build
+  nightly:
+    triggers:
+      - schedule:
+          cron: "0 0 * * *"
```
Comment (blechta) [inline .circleci/config.yml]: What timezone is it? UTC? If so maybe we should use even less busy hours.

File: .circleci/config.yml (hunk 1 above)
Comment (IgorBaratta) [reply .circleci/config.yml]: I was getting an error to persist a temporary file to the `pushdoc` job.
Changed back to master.

File: .circleci/config.yml (hunk 2 above)
Comment (IgorBaratta) [reply .circleci/config.yml]: Yes, I was thinking in changing it to `master`.
If we program all branches to run at the same time it can generate a large queue.
I haven't found a filter to run only active branches.

File: .circleci/config.yml (hunk 3 above)
Comment (IgorBaratta) [reply .circleci/config.yml]: UTC.
02:00 am would be better?

Comment (blechta) [reply .circleci/config.yml]: UTC 2.00 is still quite early for those working in CEST. What about UTC 5.00? Seems early enough before busy hours in Cambridge at winter.

File: .circleci/config.yml (hunk 2 above)
Comment (blechta) [reply .circleci/config.yml]: https://circleci.com/docs/2.0/configuration-reference/#branches-1 seems to imply that only pushed commits schedule to a build. Could you verify that experimentally?

Comment (IgorBaratta) [reply .circleci/config.yml]: Yeah, the default behavior is to trigger the workflow when pushing to a branch. With the keyword `schedule`  it behaves a bit different.
I will schedule two consecutive builds (to avoid waiting two days) without committing between them to verify if in that case a commit is needed.

Comment (IgorBaratta) [reply .circleci/config.yml]: It runs the workflow on schedule regardless of having no new commit.
See for example (hourly builds): https://circleci.com/gh/IgorBaratta/dolfinx

Comment (blechta) [reply .circleci/config.yml]: Too bad, good digging. Ok, do it the way you think is the best.

**Comment by IgorBaratta** [comment]
The complex build will not be green until [#144](https://github.com/FEniCS/dolfinx/pull/144) is merged, and the `scalar_type` form compiler parameter is set.
